test/spec/directives/bsSwitchSpec.js

```diff
@@ -672,23 +673,57 @@ describe('Directive: bsSwitch', function () {
   it('should watch updates in getterSetter', inject(makeTestGetterSetter()));
   it('should watch updates in getterSetter', inject(makeTestGetterSetter(true)));
 
-  function makeTestChange(input) {
+  function makeTestViewSwitchChange(input) {
     return function () {
-      compileDirective('change', input);
+      var element = compileDirective('change', input);
+      scope.switchChange = jasmine.createSpy();
+      scope.ngChange = jasmine.createSpy();
 
-      scope.change = function() {};
-      spyOn(scope, 'change');
+      // On
+      scope.model = true;
+      scope.$apply();
 
-      scope.model = false;
+      // Indeterminate
+      scope.model = undefined;
       scope.$apply();
-      expect(scope.change).not.toHaveBeenCalled();
+
+      // Off
+      scope.switchChange.calls.reset();
+      scope.ngChange.calls.reset();
+      element.find('input').click();
+      expect(scope.switchChange).toHaveBeenCalled();
+      expect(scope.ngChange).toHaveBeenCalled();
+
+      // On
+      scope.switchChange.calls.reset();
+      scope.ngChange.calls.reset();
+      element.find('input').click();
+      expect(scope.switchChange).toHaveBeenCalled();
+      expect(scope.ngChange).toHaveBeenCalled();
+    };
+  }
+  it('should evaluate change expression when view changes', inject(makeTestViewSwitchChange()));
+  it('should evaluate change expression when view changes', inject(makeTestViewSwitchChange(true)));
+
+  function makeTestModelSwitchChange(input) {
+    return function () {
+      compileDirective('change', input);
+      scope.switchChange = jasmine.createSpy();
+      scope.ngChange = jasmine.createSpy();
 
       scope.model = true;
       scope.$apply();
-      expect(scope.change).toHaveBeenCalled();
+      expect(scope.switchChange).not.toHaveBeenCalled();
```

Comment:
Ok, so let's keep it this way. At this point it's pointless testing the switch from `true` or `false` to `undefined`, since that could only happen via model changes and won't trigger any change function. :+1: